src/main/java/com/nashtech/exception/GlobalExceptionHandling.java

```diff
@@ -0,0 +1,44 @@
+package com.nashtech.exception;
+
+import com.azure.core.exception.ResourceNotFoundException;
+import lombok.extern.slf4j.Slf4j;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+import java.time.LocalDateTime;
+
+/**
+ * A class that handles global exception handling for all
+ * REST controllers in the application.
+ */
+@RestControllerAdvice
+@Slf4j
+public class GlobalExceptionHandling implements Thread.UncaughtExceptionHandler {
```

Comment:
Name is compling the best practices. Also, why do we need to implement this Interface?